dispatcher/src/main/kotlin/pl/droidsonroids/testing/mockwebserver/FixtureDispatcher.kt

```diff
@@ -30,6 +30,7 @@ class FixtureDispatcher internal constructor(private val responseBuilder: Respon
      * @throws IllegalArgumentException when no requests can be matched
      */
     @Throws(IllegalArgumentException::class)
+    @Synchronized
```

Comment:
I would rather wrap maps with `Collections.synchronizedSortedMap` https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#synchronizedSortedMap-java.util.SortedMap-

Reply:
Yep, it's better to convert the tree maps to thread-safe sorted maps.

I've synchronized the iterations on the maps too to make them thread-safe too.

Let me know what do you think.
Thanks for checking this ❤️
I think it's better to make all the code thread-safe correct as you suggested instead of only synchronizing the `dispatch` method.